spyder/api/plugins.py

```diff
@@ -533,3 +561,951 @@ def set_ancestor(self, ancestor):
+            ApplicationMenus.Tools: self._main.tools_menu,
+            ApplicationMenus.View: self._main.view_menu,
+            ApplicationMenus.Help: self._main.help_menu,
+        }
+
+        if name in app_menus:
+            return app_menus[name]
+        else:
+            raise Exception('Application menu "{0}" not found! Available '
+                            'menus are: {1}'.format(name,
+                                                    list(app_menus.keys())))
+
+    def get_application_toolbar(self, name):
+        """
+        Return an application toolbar by name.
+        """
+        # TODO: Temporal solution while API for managing app menus is created
+        app_toolbars = {
+            ApplicationToolBars.File: self._main.file_toolbar,
+            ApplicationToolBars.Run: self._main.run_toolbar,
+            ApplicationToolBars.Debug: self._main.debug_toolbar,
+            ApplicationToolBars.Main: self._main.main_toolbar,
+            ApplicationToolBars.Search: self._main.search_toolbar,
+            ApplicationToolBars.Edit: self._main.edit_toolbar,
+            ApplicationToolBars.Source: self._main.source_toolbar,
+        }
+        if name in app_toolbars:
+            return app_toolbars[name]
+        else:
+            raise SpyderAPIError(
+                'Application toolbar "{0}" not found! '
+                'Available toolbars are: {1}'.format(
+                    name,
+                    list(app_toolbars.keys())
+                )
+            )
+
+    def add_item_to_application_menu(self, item, menu, section=None,
+                                     before=None):
+        """
+        TODO: Menu could be a name or an instance? this could avoid calling
+        get application menu.
+        """
+        # Enable when new API is activated
+        # Check that menu is an ApplicationMenu
+        # if not isinstance(menu, ApplicationMenu):
+        #     raise SpyderAPIError('Not an ApplicationMenu!')
+
+        # TODO: For now just add to the bottom
+        #       Temporal solution while API for managing app menus is created
+        app_menu_actions = {
+            ApplicationMenus.File: self._main.file_menu_actions,
+            ApplicationMenus.Edit: self._main.edit_menu_actions,
+            ApplicationMenus.Search: self._main.search_menu_actions,
+            ApplicationMenus.Source: self._main.source_menu_actions,
+            ApplicationMenus.Run: self._main.run_menu_actions,
+            ApplicationMenus.Debug: self._main.debug_menu_actions,
+            ApplicationMenus.Consoles: self._main.consoles_menu_actions,
+            ApplicationMenus.Projects: self._main.projects_menu_actions,
+            ApplicationMenus.Tools: self._main.tools_menu_actions,
+            # ApplicationMenus.View: self._main.view_menu_actions,
+            ApplicationMenus.Help: self._main.help_menu_actions,
+        }
+        actions = app_menu_actions[menu.name]
+        actions.append(None)
```

Comment:
Once the plugins are migrated, there wont be any need for doing this, since that will be handled at the SpyderMenu level, so probably not worth it as this is temporal code.